Show anchored accordions in edit forms

replaceafill opened the pull request:
This implements a new `collapse.js` script that shows and scrolls to the accordions targeted from various index actions.

It also restricts the `js/editingHistory.js` script to load the history only from the user index page.

jraddaoui commented on plugins/arDominionB5Plugin/modules/sfIsdiahPlugin/templates/editSuccess.php:
```diff
@@ -220,7 +220,7 @@
           </button>
         </h2>
-        <div id="points-collapse" class="accordion-collapse collapse" aria-labelledby="points-heading">
+        <div id="points-collapse" class="accordion-collapse collapse" aria-labelledby="access-heading">
```
?

jraddaoui replied:
This one is `points` because there is another access area above.

replaceafill replied:
Oops, thanks for catching that :facepalm: I've fixed it in https://github.com/artefactual/atom/pull/1343/commits/e89aa566793ef92f85082e35656a46224f15aafc

jraddaoui commented on plugins/arDominionB5Plugin/modules/accession/templates/indexSuccess.php:
```diff
@@ -155,7 +155,7 @@
 <div class="section" id="rightsArea">
 
-  <?php echo link_to_if(QubitAcl::check($resource, 'update'), '<h2>'.__('Rights area').'</h2>', [$resource, 'module' => 'accession', 'action' => 'edit'], ['anchor' => 'rightsArea', 'title' => __('Edit rights area')]); ?>
+  <?php echo link_to_if(QubitAcl::check($resource, 'update'), '<h2>'.__('Rights area').'</h2>', [$resource, 'module' => 'accession', 'action' => 'edit'], ['anchor' => 'io-collapse', 'title' => __('Edit rights area')]); ?>
```
I believe this one was broken as there is no `rightsArea` in the current template either. Maybe remove this heading link and, if you want, add it to the `informationObjectArea` heading below.

replaceafill replied:
Ah I understand now. The accession I was testing with had no rights information or archival information object attached so I was not able to see those sections :+1: I think I've fixed it in https://github.com/artefactual/atom/pull/1343/commits/731774e1e1c62288aeba28794d1e918fefded26c

jraddaoui commented on plugins/arDominionB5Plugin/js/collapse.js:
```diff
@@ -0,0 +1,17 @@
+    if ($('#editForm .accordion-item').length) {
+      var $collapseToShow = $(location.hash);
+      if ($collapseToShow.length) {
+        $collapseToShow.on('shown.bs.collapse', function (e) {
```
Nitpick: arrow function.

jraddaoui commented on plugins/arDominionB5Plugin/js/collapse.js:
```diff
@@ -0,0 +1,17 @@
+        $collapseToShow.on('shown.bs.collapse', function (e) {
+          window.scrollTo(0, $(e.target).parent().offset().top);
+        });
+        bootstrap.Collapse.getOrCreateInstance($collapseToShow.get(0)).show();
```
Nitpick: `bootstrap.Collapse.getOrCreateInstance($collapseToShow);` may be enough.

replaceafill replied:
Ah I see. I thought those methods only worked with DOM objects, but they support jQuery objects too! Addressed both nitpicks in https://github.com/artefactual/atom/pull/1343/commits/830719db4ddd7345613548928f2e00fbcb28a04e

jraddaoui reviewed (changes requested):
Nicely done @replaceafill!

Glad you're going deep into B5 docs and using the event to wait for the scroll. Just two things we may want to change and a couple of nitpicks.

replaceafill commented:
Thanks you @jraddaoui. I think I've addressed your observations now.

jraddaoui commented on plugins/arDominionB5Plugin/js/collapse.js:
```diff
@@ -6,10 +6,10 @@
       var $collapseToShow = $(location.hash);
       if ($collapseToShow.length) {
-        $collapseToShow.on('shown.bs.collapse', function (e) {
+        $collapseToShow.on('shown.bs.collapse', (e) => {
```
No need of parenthesis with one argument.

jraddaoui commented on plugins/arDominionB5Plugin/js/collapse.js:
```diff
@@ -6,10 +6,10 @@
           window.scrollTo(0, $(e.target).parent().offset().top);
         });
-        bootstrap.Collapse.getOrCreateInstance($collapseToShow.get(0)).show();
+        bootstrap.Collapse.getOrCreateInstance($collapseToShow).show();
```
I think the show is not needed as the getOrCreateInstance call toggles the collapse by default.

replaceafill replied:
Nice. Thanks!

jraddaoui reviewed (approved):
Thanks @replaceafill, a couple of follow ups in the nitpicks but feel free to ignore, squash and merge.

replaceafill committed:
Show anchored accordions in edit forms

This implements a new `collapse.js` script that shows and scrolls to
the accordions targeted from various index actions.

It also restricts the `js/editingHistory.js` script to load the
history only from the user index page.
